napari/_qt/dialogs/preferences_dialog.py

```diff
@@ -230,9 +230,19 @@ def build_page_dialog(self, schema, values):
         values : dict
             Dictionary of current values set in preferences.
         """
-
         builder = WidgetBuilder()
         form = builder.create_form(schema, self.ui_schema)
+
+        # Disable widgets that loaded settings from environment variables
+        section = schema["section"]
+        form_layout = form.widget.layout()
+        for row in range(form.widget.layout().rowCount()):
+            widget = form_layout.itemAt(row, form_layout.FieldRole).widget()
+            name = widget._name
+            widget.setDisabled(
```

Comment:
This line isn't working for me.  If I do something like the following it works as expected: 

```
if SETTINGS._env_settings.get(section, {}).get(name, False) is not False:
                widget.setDisabled(True)
```

Reply:
Will check @ppwadhwa, thanks

Reply:
Fixed!